### PR TITLE
Correct unused parameter warnings in replace algorithms

### PR DESCRIPTION
**Pull request by robertmaynard**
Starting in CUDA 11.3, nvcc will start to unconditionally warn about unused parameters on functions/methods that are in anonymous namespaces.

### CODE REVIEW

**Review by karthikeyann — APPROVED** [review]

**Comment by robertmaynard** [comment]
@gpucibot merge

File: cpp/src/replace/nulls.cu [hunk 1]
```diff
@@ -367,7 +366,7 @@ std::unique_ptr<cudf::column> replace_nulls_scalar_kernel_forwarder::operator()<
 std::unique_ptr<cudf::column> replace_nulls_policy_impl(cudf::column_view const& input,
                                                         cudf::replace_policy const& replace_policy,
                                                         rmm::cuda_stream_view stream,
-                                                        rmm::mr::device_memory_resource* mr)
+                                                        rmm::mr::device_memory_resource*)
```
Comment (cwharris) [inline cpp/src/replace/nulls.cu:369]: this is a bug. we should be using the provided mr to allocate the return value.

Comment (robertmaynard) [reply cpp/src/replace/nulls.cu:369]: Yea, it should be propagated down to the `cudf::detail::gather` call. I will open a PR that fixes this issue

Comment (robertmaynard) [reply cpp/src/replace/nulls.cu:369]: https://github.com/rapidsai/cudf/pull/8500

**Pull request merged**